botocore/httpchecksum.py

```diff
@@ -355,6 +356,17 @@ def apply_request_checksum(request):
     checksum_context = request.get("context", {}).get("checksum", {})
     algorithm = checksum_context.get("request_algorithm")
 
+    if "method" in request and (
```

Comment:
I have a few questions about the logic here:
- Why is this being guarded by a conditional checking for the `HEAD` and `GET` HTTP methods? Do we only want to track this metric in cases where response checksum validation would occur? My understanding is we want to always track this metric, despite what operations are being used.
- This method is responsible for applying a flexible checksum to a request. To me it doesn't make sense to put this logic here since it's related to response checksum config. I think this makes more sense to go where we determine the config values for `response_checksum_validation` in `args.py` if possible.